cmd/system-probe/modules/network_tracer.go

```diff
@@ -80,11 +99,89 @@ type networkTracer struct {
 	restartTimer *time.Timer
 }
 
+func startRPCServer(tracer networkTracer) error {
+	socketFile := "/tmp/my_grpc.sock"
+
+	// Remove existing socket file if it exists
+	os.Remove(socketFile)
+
+	listener, err := net.Listen("unix", socketFile)
+	if err != nil {
+		return err
+	}
+
+	server := grpc.NewServer()
+	test2.RegisterSystemProbeServer(server, &tracer)
+
+	useGRPCServer := dd_config.SystemProbe.GetBool("service_monitoring_config.use_grpc")
+	if useGRPCServer {
+		go func() {
+			log.Info("gRPC server listening on Unix domain socket...")
+
+			err = server.Serve(listener)
+			if err != nil {
+				return
+			}
+		}()
+	}
+
+	// Wait for interrupt signal to gracefully stop the server
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, os.Interrupt, syscall.SIGTERM)
+	<-sigCh
+
+	// Gracefully stop the server
+	server.GracefulStop()
+	return nil
+}
+
 func (nt *networkTracer) GetStats() map[string]interface{} {
 	stats, _ := nt.tracer.GetStats()
 	return stats
 }
 
+func writeConnectionsTogRPC(marshaler encoding.Marshaler, cs *network.Connections) ([]byte, error) {
+	defer network.Reclaim(cs)
+
+	buf, err := marshaler.Marshal(cs)
+	if err != nil {
+		log.Errorf("unable to marshall connections with type %s: %s", marshaler.ContentType(), err)
+		return nil, err
+	}
+
+	log.Tracef("/connections: %d connections, %d bytes", len(cs.Conns), len(buf))
+	return buf, nil
+}
+
+func (nt *networkTracer) GetConnections(req *test2.GetConnectionsRequest, s2 test2.SystemProbe_GetConnectionsServer) error {
+	start := time.Now()
+	var runCounter = atomic.NewUint64(0)
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏
```suggestion
	runCounter := atomic.NewUint64(0)
```